File: src/main/java/net/softbell/bsh/dto/request/IotReservDto.java
package net.softbell.bsh.dto.request;

import java.util.HashMap;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 예약정보 DTO
 */
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class IotReservDto
{
    private boolean enableStatus;
    private String description;
    private String expression;
    private HashMap<Long, IotActionDto> mapAction;
}

File: src/main/kotlin/net/softbell/bsh/dto/request/IotReservDto.kt
package net.softbell.bsh.dto.request

/**
 * @author : Bell(bell@softbell.net)
 * @description : 예약정보 DTO
 */
data class IotReservDto (
        var enableStatus: Boolean,
        var description: String,
        var expression: String,
        var mapAction: HashMap<Long, IotActionDto>? // TODO Not Null 대체방법 강구
)